plugins/inputs/qbittorrent/qbittorrent.go

```diff
@@ -0,0 +1,222 @@
+package qbittorrent
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+var globalMainData *MainData
+
+type QBittorrent struct {
+	URL      string        `toml:"url"`
+	Username config.Secret `toml:"username"`
+	Password config.Secret `toml:"password"`
+
+	cookie []*http.Cookie
+}
+
+func (q *QBittorrent) Init() error {
+	if q.Username.Empty() && q.Password.Empty() {
+		return errors.New("username and password is empty")
+	}
+	return nil
+}
+
+func (*QBittorrent) SampleConfig() string {
+	return sampleConfig
+}
+
+// Gather ..
+func (q *QBittorrent) Gather(acc telegraf.Accumulator) error {
+	err := q.getSyncData()
+	if err != nil {
+		return err
+	}
+	for k, v := range globalMainData.toMetrics() {
+		for i := range v {
+			acc.AddFields(k, v[i].Fields(), v[i].Tags())
+		}
+	}
+
+	return nil
+}
```

Comment:
A bit complex, You could let the `getSyncData` return the metrics array and then just do `acc.AddMetric` on each..